pgjdbc/src/main/java/org/postgresql/jdbc/TimestampUtils.java

```diff
@@ -360,40 +361,67 @@ private ParsedTimestamp parseBackendTimestamp(String str) throws SQLException {
 
     } catch (NumberFormatException nfe) {
       throw new PSQLException(
-          GT.tr("Bad value for type timestamp/date/time: {0}", str),
+          GT.tr("Bad value for type timestamp/date/time: {0}", new String(s)),
```

Comment:
`new String(s)` relies on the Java default encoding which does not necessarily matches DB encoding. I guess we need something like `queryExecutor.getEncoding().decode(s)`